ecs-init/docker/docker.go

```diff
@@ -541,3 +576,32 @@ func (c *client) StopAgent() error {
 	}
 	return err
 }
+
+// isDomainJoined is used to validate if container instance is part of a valid active directory.
+func isDomainJoined() bool {
```

Comment:
nit: can you add a unit test for this method? (you would need to mock `exec` methods)
example: https://github.com/aws/amazon-ecs-agent/blob/71a0f6765cc469913a0b09728196feb0d0d33e37/agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows_test.go#L395

Reply:
added tests